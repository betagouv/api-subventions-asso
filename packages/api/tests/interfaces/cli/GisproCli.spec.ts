import GisproCli from "../../../src/interfaces/cli/Gispro.cli";
import gisproPort from "../../../src/dataProviders/db/providers/gispro.port";
import path from "path";

describe("Gispro Cli", () => {
    const cli = new GisproCli();
    const FILE_PATH = path.resolve(__dirname, "./__fixtures__/2023 subventions GISPO et TDB_test.xlsx");

    const EXPORT_DATE_STR = "2023-06-06";

    describe("parse", () => {
        it("saves uniformized data from 2023", async () => {
            await cli.parse(FILE_PATH, EXPORT_DATE_STR);
            const savedData = await gisproPort.findAll();
            const noId = savedData.map(({ _id, ...rest }) => rest);
            expect(noId).toMatchSnapshot();
        });
    });
});
