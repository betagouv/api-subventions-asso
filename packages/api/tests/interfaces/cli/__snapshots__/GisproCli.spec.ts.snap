// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gispro Cli parse saves uniformized data from 2023 1`] = `
[
  {
    "codeActionDossier": "00236035",
    "codeProjet": "7665P00119",
    "directionGestionnaire": "Hautes-Pyrénées",
    "ej": "2104093248",
    "exercise": 2023,
    "montant": 1200,
    "siret": "41177507500046",
    "typeBeneficiaire": "Association",
    "typeProcedure": "Annuel",
  },
]
`;
